vcd/resource_vcd_nsxt_edgegateway_dns_test.go

```diff
@@ -0,0 +1,586 @@
+    ]
+
+    domain_names = [
+      "{{.DomainName1}}",
+      "{{.DomainName2}}",
+      "{{.DomainName3}}",
+    ]
+  }
+}
+
+resource "vcd_nsxt_edgegateway_dns" "{{.VdcGroupDnsConfig}}" {
+  edge_gateway_id = data.vcd_nsxt_edgegateway.{{.VdcGroupEdgeGw}}.id
+  enabled         = true
+  
+  default_forwarder_zone {
+    name = "{{.DefaultForwarderName}}"
+    
+    upstream_servers = [
+      "{{.ServerIp1}}",
+      "{{.ServerIp2}}",
+      "{{.ServerIp3}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone1}}"
+
+    upstream_servers = [
+      "{{.ServerIp4}}",
+      "{{.ServerIp5}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName1}}",
+      "{{.DomainName2}}",
+      "{{.DomainName3}}",
+    ]
+  }
+}
+`
+
+const testAccVcdNsxtEdgegatewayDnsStep2 = testAccVcdNsxtEdgegatewayDnsPrereqs + `
+resource "vcd_nsxt_edgegateway_dns" "{{.DnsConfig}}" {
+  edge_gateway_id = data.vcd_nsxt_edgegateway.{{.EdgeGw}}.id
+  enabled         = true
+  
+  default_forwarder_zone {
+    name = "{{.DefaultForwarderName}}"
+    
+    upstream_servers = [
+      "{{.ServerIp1}}",
+      "{{.ServerIp2}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone1}}"
+
+    upstream_servers = [
+      "{{.ServerIp5}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName2}}",
+      "{{.DomainName3}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone2}}"
+
+    upstream_servers = [
+      "{{.ServerIp3}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName1}}",
+    ]
+  }
+}
+
+data "vcd_nsxt_edgegateway_dns" "data_{{.DnsConfig}}" {
+  edge_gateway_id = vcd_nsxt_edgegateway_dns.{{.DnsConfig}}.edge_gateway_id
+
+  depends_on = [vcd_nsxt_edgegateway_dns.{{.DnsConfig}}]
+}
+
+resource "vcd_nsxt_edgegateway_dns" "{{.VdcGroupDnsConfig}}" {
+  edge_gateway_id = data.vcd_nsxt_edgegateway.{{.EdgeGw}}.id
+  enabled         = true
+  
+  default_forwarder_zone {
+    name = "{{.DefaultForwarderName}}"
+    
+    upstream_servers = [
+      "{{.ServerIp1}}",
+      "{{.ServerIp2}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone1}}"
+
+    upstream_servers = [
+      "{{.ServerIp5}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName2}}",
+      "{{.DomainName3}}",
+    ]
+  }
+
+  conditional_forwarder_zone {
+    name = "{{.ConditionalForwardZone2}}"
+
+    upstream_servers = [
+      "{{.ServerIp3}}",
+    ]
+
+    domain_names = [
+      "{{.DomainName1}}",
+    ]
+  }
+}
+
+data "vcd_nsxt_edgegateway_dns" "data_{{.VdcGroupDnsConfig}}" {
+  edge_gateway_id = vcd_nsxt_edgegateway_dns.{{.VdcGroupDnsConfig}}.edge_gateway_id
+
+  depends_on = [vcd_nsxt_edgegateway_dns.{{.VdcGroupDnsConfig}}]
+}
+`
+
+func TestAccVcdNsxtEdgegatewayDnsIpSpaces(t *testing.T) {
+	preTestChecks(t)
+
+	if checkVersion(testConfig.Provider.ApiVersion, "<38.0") {
+		t.Skip("This test is only supported since version 38.1 of the API")
```

Comment:
The comment does not match the check itself

Reply:
You're right, fixed.